### ISSUE TITLE
Crash related to missing VCRUNTIME140.dll

### ISSUE THREAD

**Issue by GoogleFrog** [beyond-all-reason/spring#184]
ruru reported an unusual bug. I was not able to get an infolog out of them and I don't understand how they were able to chat in the lobby. They said this

> [00:23] ruru: solved in ruru's computer with installation of "vc_redist.x64.exe" >> Visual C++ Redistributable for Visual Studio 2015
[00:26] ruru: the problem was: "VCRUNTIME140.dll is missing" cant launch "spring.exe"
[00:27] ruru: I rethank you for your time in ZK.
[00:31] ruru is now offline

Maybe something related to debug was left in, or a missing dll from the package?

**Comment by lhog** [comment]
I had to fall back to two MSCV libs to enable https://github.com/beyond-all-reason/spring/issues/87
Now this is striking back apparently.

**Comment by lhog** [comment]
How many people have reported this?
Building FontConfig and FreeType under MinGW is expected to take some time. I'll prioritize according to number of affected people.

**Comment by GoogleFrog** [comment]
One report, but more crashes exist than reports generally.

I'm not willing to pay much for building font config, either some low number of reports or a long loading time for anyone.

**Comment by GoogleFrog** [comment]
Another crash linked above. It took a bit of back and forth to determine this was the VCRUNTIME problem. I now have an exit code, so here is an upper bound for the frequency of the issue

https://github.com/ZeroK-RTS/CrashReports/issues?q=is%3Aissue+is%3Aopen+1073741515%2C

![image](https://user-images.githubusercontent.com/1318249/150670744-8f4ee9a9-1842-47b9-98e0-4c8d6a557a93.png)

**Comment by GoogleFrog** [comment]
> How many people have reported this?
Building FontConfig and FreeType under MinGW is expected to take some time. I'll prioritize according to number of affected people.

We should be clear on whose priorities we're taking into account and who is paying the cost. I don't care about building the font config. My preference would be for it to not crash anyone, and if it noticeably impacts loading time, to turn it off. Two confirmed crashes out of about 1400 players on 798 so far is too many for me. I expect more people are affected since reporting this takes effort.

**Comment by lhog** [comment]
I updated the libs from 3rd party mingw distribution MSYS2. https://github.com/beyond-all-reason/mingwlibs64/commit/6df085ff189883dbef589d34ea1623cd22d2d8b2

Works for me here, but I have all these dll packs installed. Please check if https://github.com/beyond-all-reason/spring/actions/runs/1736119895 (when done) works for you and testers.

**Comment by lhog** [comment]
Updated libs once again to rectify annoying files like `D:\a_temp\msys\msys64\var\cache\fontconfig\31cef28e4a2b42a5a005ac7a04907b7d-x86_64.cache-7` appearing not where expected. Now they should appear in current dir, whatever it means. Need to test how it works obviously, because distribution was changed once again from MSYS2 to MXE.

I've also requested that fontconfig folks added API to set cache path explicitly, currently it's only possible via configuration (in short not on Windows).
